Decompiler/Decompiler.cs

```diff
@@ -420,11 +420,20 @@ private void ProcessFile(string path, Stream stream, string originalPath = null)
 
                 if (OutputFile != null)
                 {
-                    var data = FileExtract.Extract(resource);
+                    var contentFile = FileExtract.Extract(resource);
 
-                    var filePath = Path.ChangeExtension(path, extension);
+                    path = Path.ChangeExtension(path, extension);
+                    var outFilePath = GetOutputPath(path);
 
-                    DumpFile(filePath, data);
+                    var extensionNew = Path.GetExtension(outFilePath);
+                    if (extensionNew.Length == 0 || (extensionNew[1..]) != extension)
+                    {
+                        Console.ForegroundColor = ConsoleColor.Yellow;
+                        Console.WriteLine($"Extension `.{extension}` might be more suitable than the one provided `{extensionNew}`");
```

Comment:
Lock `ConsoleWriterLock`, and use `'` instead of ` for consistency with other messages.